### PR TITLE
Docker: add xfs tests

### PR DESCRIPTION
**Pull request by llugin**
- xfs integrated with docker images
- added patches to work with pmemfile
- xfs integrated with existing test scripts

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemfile/380)

### CODE REVIEW

File: utils/docker/images/Dockerfile.fedora-23 [hunk 1]
```diff
@@ -40,6 +40,7 @@ MAINTAINER marcin.slusarz@intel.com
 
 # Install basic tools
 RUN dnf install -y \
+	attr.x86_64 \
```
Comment (marcinslusarz) [inline utils/docker/images/Dockerfile.fedora-23]: There is no need to specify architecture.

**Comment by llugin** [comment]
Review status: 0 of 17 files reviewed at latest revision, 1 unresolved discussion.

---

*[utils/docker/images/Dockerfile.fedora-23, line 43 at r2](https://reviewable.io:443/reviews/pmem/pmemfile/380#-KuiZjXjep1U9JR7Y8RQ-r2-43:-KutAx-mzl_MBKyyR9EL:b-896fix) ([raw file](https://github.com/pmem/pmemfile/blob/d8e7e2c4eda1e40bb66c0850d85e06b83e522f87/utils/docker/images/Dockerfile.fedora-23#L43)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

There is no need to specify architecture.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/380)*

**Comment by marcinslusarz** [comment]
Reviewed 1 of 10 files at r1, 6 of 8 files at r2, 7 of 10 files at r3.
Review status: 13 of 16 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[utils/docker/external_tests/run-build-suite.sh, line 56 at r3](https://reviewable.io:443/reviews/pmem/pmemfile/380#-KuyEDcU_MfyLRCe5-Zo:-KuyEDcU_MfyLRCe5-Zp:bpkhd4h) ([raw file](https://github.com/pmem/pmemfile/blob/dd1d9c10d862deff1fab6bae3f634d800db843fe/utils/docker/external_tests/run-build-suite.sh#L56)):*
> ```Shell
> SUITE=$1
> 
> SUITE_HOME=/home/user
> ```

Why not set SUITE_HOME to $HOME? 

---

*[utils/docker/images/Dockerfile.fedora-23, line 43 at r2](https://reviewable.io:443/reviews/pmem/pmemfile/380#-KuiZjXjep1U9JR7Y8RQ-r2-43:-KuyDuFsS8KpoibnYdH1:b-gmy2id) ([raw file](https://github.com/pmem/pmemfile/blob/d8e7e2c4eda1e40bb66c0850d85e06b83e522f87/utils/docker/images/Dockerfile.fedora-23#L43)):*
<details><summary><i>Previously, llugin wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Do I really have to point out each line where it wasn't "done"...?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/380)*

**Comment by llugin** [comment]
Review status: 13 of 16 files reviewed at latest revision, 2 unresolved discussions.

---

*[utils/docker/external_tests/run-build-suite.sh, line 56 at r3](https://reviewable.io:443/reviews/pmem/pmemfile/380#-KuyEDcU_MfyLRCe5-Zo:-KuyHJ7IaxVNXYuHldqn:b-xozkn5) ([raw file](https://github.com/pmem/pmemfile/blob/dd1d9c10d862deff1fab6bae3f634d800db843fe/utils/docker/external_tests/run-build-suite.sh#L56)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Why not set SUITE_HOME to $HOME? 
</blockquote></details>

suites are installed by user, while xfs uses pmemfile-mount so it has to be run as root - in this case suites won't be found under $HOME

---

*[utils/docker/images/Dockerfile.fedora-23, line 43 at r2](https://reviewable.io:443/reviews/pmem/pmemfile/380#-KuiZjXjep1U9JR7Y8RQ-r2-43:-KuyKOU4AjgCihvB1SfX:b-896fix) ([raw file](https://github.com/pmem/pmemfile/blob/d8e7e2c4eda1e40bb66c0850d85e06b83e522f87/utils/docker/images/Dockerfile.fedora-23#L43)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Do I really have to point out each line where it wasn't "done"...?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/380)*

**Comment by krzycz** [comment]
Reviewed 1 of 10 files at r1, 6 of 8 files at r2, 7 of 10 files at r3, 3 of 3 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/380)*

**Comment by marcinslusarz** [comment]
:lgtm:

---

Reviewed 3 of 3 files at r4.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/380#-:-KuyUN2wGEkMceY0p7OB:bnfp4nl)*

**Comment by sarahjelinek** [comment]
:lgtm:

---

Reviewed 1 of 10 files at r1, 6 of 8 files at r2, 7 of 10 files at r3, 3 of 3 files at r4.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/380#-:-KuzGH4ltr5CO19wLUaF:bnfp4nl)*

**Pull request merged**